app/retail/views.py

```diff
@@ -538,6 +544,78 @@ def browser_extension_firefox(request):
     return redirect('https://addons.mozilla.org/en-US/firefox/addon/gitcoin/')
 
 
+def idea_show(request, idea_id):
+    return TemplateResponse(request, 'idea.html', {})
+
+
+def new_idea(request):
+    if request.POST:
```

Comment:
if post create idea and redirect to created idea, otherwise show idea entry form.
Is this what You were asking about ?I can also use ModelForm :)